Userland/DevTools/HackStudio/Git/GitWidget.cpp

```diff
@@ -139,13 +140,13 @@ void GitWidget::unstage_file(const LexicalPath& file)
 
 void GitWidget::commit()
 {
-    String message;
-    auto res = GUI::InputBox::show(window(), message, "Commit message:", "Commit");
-    if (res != GUI::InputBox::ExecOK || message.is_empty())
-        return;
-    dbgln("commit message: {}", message);
-    m_git_repo->commit(message);
-    refresh();
+    auto dialog = GitCommitDialog::construct(window());
+
+    dialog->on_commit = [this](auto& message) {
+        m_git_repo->commit(message);
+        refresh();
+    };
+    dialog->exec();
```

Comment:
Continuing from my upper comment about the `on_commit` handler, maybe do not initialize the dialog at all if there is no git repository, and show an error message instead.